### PR TITLE
makefile: Add making ota img functionality for artik05x

### PR DESCRIPTION
**Pull request by jeongchanKim**
functionality for attaching crc and signing to tinyara binary

### CODE REVIEW

**Comment by seinfra** [comment]
Code Rule Check OK.

File: build/configs/artik053/audio/Make.defs [hunk 1]
```diff
@@ -155,3 +155,7 @@ define DOWNLOAD
   @$(TOPDIR)/../build/configs/artik05x/artik05x_download.sh --board=artik053 $(1) $(2) $(3) $(4) $(5) $(6)
 endef
 
+define OTA_IMG
+  $(Q) $(TOPDIR)/../build/configs/artik05x/artik05x_ota.sh \
+    --board=artik053 --secure ota
```
Comment (sunghan-chang) [inline build/configs/artik053/audio/Make.defs]: 1. Is ```--secure``` needed to normal model?
2. Let's make it a line. We don't need to change a line.

File: build/configs/artik05x/artik05x_ota.sh [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+# either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+#
+###########################################################################
+#
+# File   : artik05x_ota.sh
+# Description : Attach CRC and Signing in TizenRT Binary
+
+# Remember, make is invoked from "os" directory
+source .config
```
Comment (sunghan-chang) [inline build/configs/artik05x/artik05x_ota.sh]: ```CONFIG_XX``` is not used in this file. Let's remove this.

File: build/configs/artik05x/artik05x_ota.sh [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    local bin=
+    test -e $TIZENRT_BIN-signed && rm -rf $TIZENRT_BIN-signed
+    bin=$(make-target-bin --target=$TARGET_NAME --bin=$TARGET_NAME)
+}
+
+if test $# -eq 0; then
+    exit 1
+fi
+
+while test $# -gt 0; do
```
Comment (sunghan-chang) [inline build/configs/artik05x/artik05x_ota.sh]: You don't use argument ```ota```.  Why do you give it in Make.defs?

File: build/configs/artik05x/artik05x_ota.sh [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+    echo $bin
+
+    return 0
+}
+
+ota_crc() {
+    local strcmd=""
+    local bin=
+    test -e $TIZENRT_BIN-signed && rm -rf $TIZENRT_BIN-signed
```
Comment (sunghan-chang) [inline build/configs/artik05x/artik05x_ota.sh]: After executing ```signing```, ```TIZENRT_BIN``` is tinyara_head.bin-signed. What is ```$TIZENRT_BIN-signed```? It looks wrong.

File: build/configs/artik05x/artik05x_ota.sh [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+    bin=$(make-target-bin --target=$TARGET_NAME --bin=$TARGET_NAME)
```
Comment (sunghan-chang) [inline build/configs/artik05x/artik05x_ota.sh]: Why do you save ```bin```? What purpose?

File: build/configs/artik05x/artik05x_ota.sh [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+            test ! -e $bin && return 1 ;;
+    esac
+
+    echo $bin
+
+    return 0
+}
+
+ota_crc() {
+    local strcmd=""
```
Comment (sunghan-chang) [inline build/configs/artik05x/artik05x_ota.sh]: Not used.

File: build/configs/artik05x/artik05x_ota.sh (hunk 3 above)
Comment (jeongchanKim) [reply build/configs/artik05x/artik05x_ota.sh]: modified.

**Comment by seinfra** [comment]
Code Rule Check OK.

**Pull request merged**